spec/map_spec.js

```diff
@@ -0,0 +1,25 @@
+import { map } from '../src/map'
+
+var users = [
+  { 'user': 'barney' },
+  { 'user': 'fred' }
+];
+
+
+describe('intersection', ()=> {
+  it('returns the intersection of values present in all input arrays', () => {
+    expect(map([1, 2, 3, 4], (n) => {return n * n})).toEqual([1, 4, 9, 16])
```

Comment:
ES6 allows us to make this more concise:
```es6
expect( map( [1, 2, 3, 4], n = > n * n ) ).toEqual( [1, 4, 9, 16 ] )
```
Recall that in ES6, a function that takes only one argument may omit the parentheses, and a function that has only a single statement in the body that is the return statement may omit the curly braces.